### ISSUE TITLE
`sugar.collect` can't be used in templates

### ISSUE THREAD

**Issue by dawkot** [nim-lang/Nim#14332]
### Example
```nim
import sugar

template foo =
  discard collect(newSeq, for i in 1..3: i)

foo()
```

### Current Output
```
Error: type mismatch: got <>
        ... but expected one of: 
        ... proc (s: var seq[int], len: Natural){.noSideEffect.}
```
### Additional Information

* Nim 1.2.0

**Comment by Araq** [comment]
Workaround:

```nim

import sugar

template foo {.dirty.} =
  discard collect(newSeq, for i in 1..3: i)

foo()


```

**Comment by planetis-m** [comment]
Fixed by #16089
